docker/Dockerfile.build

```diff
@@ -1,56 +1,85 @@
-# Builder image
-FROM rust:buster as builder
+# This steps implement cargo-chef for docker layer caching
+# We are using four stages:
+# - chef: installs cargo-chef
+# - planner: computes the recipe file
+# - builder: caches our dependencies and builds the binary
+# - runtime: is our runtime environment
+FROM rust:1.58.1-bullseye as chef
+RUN cargo install cargo-chef
+WORKDIR /app
+
+FROM chef AS planner
+COPY . .
+RUN cargo chef prepare  --recipe-path recipe.json
 
+FROM chef AS builder
+COPY --from=planner /app/recipe.json recipe.json
 RUN apt-get update && \
-	apt-get install -y --no-install-recommends \
-	make cmake g++ gcc llvm libclang-dev clang ca-certificates
-
-RUN mkdir /zebra
-WORKDIR /zebra
+    apt-get install -y --no-install-recommends \
+    make cmake g++ gcc llvm libclang-dev clang ca-certificates && \
+    rm -Rf /var/lib/apt/lists/* /tmp/*
+# Build dependencies - this is the caching Docker layer!
+RUN cargo chef cook --release --recipe-path recipe.json
 
+# Build zebra
 ARG SHORT_SHA
 ENV SHORT_SHA $SHORT_SHA
-
-ENV RUST_BACKTRACE full
-ENV CARGO_HOME /zebra/.cargo/
+ARG RUST_BACKTRACE
+ENV RUST_BACKTRACE ${RUST_BACKTRACE:-full}
+# Skip test on debian based OS by default as compilation fails
```

Comment:
This test failure is a bit of a surprise. The tests worked with the previous dockerfile.

Have you checked to see if the network tests are still failing?
Have you tried `ZEBRA_SKIP_IPV6_TESTS` instead?

(I'm not sure I understood the comment correctly, but let's see if it's still needed first.)

Reply:
I made this change to have the same behavior as we have in the CI, and standardize it. https://github.com/ZcashFoundation/zebra/blob/main/.github/workflows/ci.yml#L63

Maybe that was a specific problem in GitHub Actions.

Reply:
Changed to `ZEBRA_SKIP_IPV6_TESTS` as we don't have IPv6 available in GHA